Enable/View Controllers/LoginViewController.m

```diff
@@ -51,30 +51,28 @@ - (IBAction)didTapSignUp:(id)sender {
     userProfile.username = @"Anonymous User";
     userProfile.email = email;
     
-    if(![self isEmail:email]){
+    if([self isEmail:email]){
+        [user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError *error) {
+            if (!error) {
+                userProfile.userID = [PFUser currentUser];
+                [userProfile saveInBackgroundWithBlock:^(BOOL saveSucceeded, NSError * _Nullable saveError) {
+                    if(!saveError){
+                        [self navigateToProfile];
+                    } else {
+                        NSLog(@"%@", saveError.localizedDescription);
+                    }
+                }];
+            } else {
+                NSLog(@"%@", error.localizedDescription);
+            }
+        }];
+    } else {
         NSLog(@"NOT AN EMAIL !!!!!!!!!!!!!");
-        return;
     }
-    [user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError *error) {
-        if (!error) {
-            userProfile.userID = [PFUser currentUser];
-            [userProfile saveInBackgroundWithBlock:^(BOOL saveSucceeded, NSError * _Nullable saveError) {
-                if(!saveError){
-                    [self navigateToProfile];
-                }
-                else {
-                    NSLog(@"issues!");
-                    NSLog(@"%@", saveError.localizedDescription);
-                }
-            }];
-        } else {
-            NSLog(@"👿👿👿👿 doesn't work");
-            NSLog(@"%@", error.localizedDescription);
-        }
-    }];
+
 }
 
-- (IBAction)didTapLogin:(id)sender {
+- (void) logIn {
     [PFUser logInWithUsernameInBackground:self.emailTextField.text password:self.passTextField.text block:^(PFUser* user, NSError* error){
```

Comment:
When I'm talking about separating the UI logic and the business logic I'm thinking more at something like this:
* UI logic: get the email and password from the text fields
* Business logic: perform login for a given email and password
```
- (IBAction)didTapLogin:(id)sender {
    [self loginWith:self.emailTextField.text and:self.passTextField.text];
}

- (void) loginWith: (NSString *)email and:(NSString *)password {
    [PFUser logInWithUsernameInBackground:email password:password block:^(PFUser* user, NSError* error){
            if(!error){
                NSLog(@"success");
    }];
}
```
What I was trying to explain for testing is, when you have some code that looks like this, it is very easy to test the `login` function, passing any kind of email/password as a parameter of the function AND in the other hand it is much more complicated if the function is getting input from the UI (here text fields).